.github/workflows/cache.yml

```
@@ -0,0 +1,43 @@
+name: Cache
+on:
+  push:
+    branches: [develop]
+    paths:
+      - '.github/**'
+      - 'tools/buildsys/**'
+      - 'tools/pubsys*/**'
+      - 'tools/pubsys*/**'
+      - '!tools/pubsys/policies/**'
+      - '!tools/pubsys/**.example'
+      - '!tools/pubsys/**.template'
+      - 'tools/Cargo.lock'
+jobs:
+  cache:
+    runs-on: [self-hosted, linux, x64]
+    continue-on-error: true
+    steps:
+      - run: |
+            echo "OS_ARCH=`uname -m`" >> $GITHUB_ENV
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo
+            tools/bin
+            tools/.crates.toml
+            tools/.crates2.json
+            tools/target
+          # You can edit the .github/cache_bust file if you need to clear the cache
+          key: ${{ hashFiles('.github/cache_bust') }}-${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('tools/Cargo.lock') }}
```

Instead of having a complete cache-miss when cargo.lock changes, you can use a previous cache and then update the cache like this.

```suggestion
          key: ${{ hashFiles('.github/cache_bust') }}-${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}-${{ hashFiles('tools/Cargo.lock') }}
        restore-keys: |
          key: ${{ hashFiles('.github/cache_bust') }}-${{ hashFiles('.github/workflows/cache.yml') }}-${{ runner.os }}-${{ env.OS_ARCH }}-
```